src/pages/VideoCall/permissions/Check.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import Icon from '@mdi/react'
+import { mdiCheckCircle, mdiAlertCircle } from '@mdi/js'
+import styled from 'styled-components'
+import { Spinner } from 'react-bootstrap'
+import {
+    STATE_DENIED,
+    STATE_GRANTED,
+    STATE_WAITING,
+} from '../../../hooks/useCameraMicrophonePermission'
+
+export const CheckItem = styled.div`
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    svg {
+        margin-right: 8px;
+    }
+
+    .spinner {
+        flex-shrink: 0;
+        flex-basis: auto;
+        margin-right: 12px;
+        margin-left: 4px;
+        width: 30px;
+        height: 30px;
+    }
+`
+
+const iconSize = 1.5
+const Check = ({ state, i18nKey, i18nInstance }) => {
```

Comment:
you can import i18n from '../../../i18n/i18n' and use it like this. That avoid you to inject i18nInstance
```i18n.t(`videocall:${i18nKey}.waiting`)```

Reply:
you're right, Check was abtract to any logic but I've changed my mind in the meantime.